cpp/src/arrow/compute/exec/expression_test.cc

```diff
@@ -927,6 +927,26 @@ TEST(Expression, FoldConstantsBoolean) {
   ExpectFoldsTo(or_(whatever, whatever), whatever);
 }
 
+struct {
+  void operator()(Expression expr, Expression expected,
+                  const Schema& schema = *kBoringSchema) {
+    ASSERT_OK_AND_ASSIGN(expr, expr.Bind(schema));
+    ASSERT_OK_AND_ASSIGN(expected, expected.Bind(schema));
+
+    ASSERT_OK_AND_ASSIGN(auto without_named_refs, RemoveNamedRefs(expr));
+
+    EXPECT_EQ(without_named_refs, expected);
+  }
+} ExpectRemovesRefsTo;
```

Comment:
Am I missing something that makes this function better/more useful than a simple function?

Reply:
No, I was copying `ExpectFoldsTo` but I see there are other `ExpectsXyz` methods in this file which aren't structs and so I changed it to a plain method for simplicity.